### PR TITLE
support new mgmt channel info struct format

### PR DESCRIPTION
**Pull request by mamin506**

### CODE REVIEW

File: src/driver/amdxdna/aie2_pci.c [hunk 1]
```diff
@@ -54,11 +78,56 @@ static inline void aie2_dump_chann_info_debug(struct amdxdna_dev_hdl *ndev)
...
+	XDNA_DBG(xdna, "mailbox protocol minor 0x%x", ndev->mgmt_prot_minor);
+}
+
+static bool aie2_is_legacy_mgmt_info(struct amdxdna_dev_hdl *ndev, u32 off)
+{
+	u32 *mgmt_info_bar_addr = ndev->sram_base + off;
+	u32 magic;
+
+	magic = readl(mgmt_info_bar_addr + MAGIC_OFFSET);
+	if (strcmp((const char *)&magic, MGMT_MBOX_MAGIC))
```
Comment (maxzhen) [inline src/driver/amdxdna/aie2_pci.c]: strcmp? Why not just define MGMT_MBOX_MAGIC as u32 and do "=="?

**Review by maxzhen — CHANGES_REQUESTED** [review]

File: src/driver/amdxdna/aie2_pci.c [hunk 2]
```diff
@@ -54,11 +78,56 @@ static inline void aie2_dump_chann_info_debug(struct amdxdna_dev_hdl *ndev)
...
+
+	i2x->mb_head_ptr_reg = AIE2_MBOX_OFF(ndev, info_regs.i2x_head);
+	i2x->mb_tail_ptr_reg = AIE2_MBOX_OFF(ndev, info_regs.i2x_tail);
+	i2x->rb_start_addr   = AIE2_SRAM_OFF(ndev, info_regs.i2x_buf);
+	i2x->rb_size         = info_regs.i2x_buf_sz;
+
+	x2i->mb_head_ptr_reg = AIE2_MBOX_OFF(ndev, info_regs.x2i_head);
+	x2i->mb_tail_ptr_reg = AIE2_MBOX_OFF(ndev, info_regs.x2i_tail);
+	x2i->rb_start_addr   = AIE2_SRAM_OFF(ndev, info_regs.x2i_buf);
+	x2i->rb_size         = info_regs.x2i_buf_sz;
```
Comment (maxzhen) [inline src/driver/amdxdna/aie2_pci.c]: pls do not duplicate these lines of code. just fill them in as before and when you reach the end, read the magic u32 to decide if you need to read more or not. all of these can be done in one function.

File: src/driver/amdxdna/aie2_pci.c (hunk 1 above)
Comment (mamin506) [reply src/driver/amdxdna/aie2_pci.c]: Thanks. Fixed.

File: src/driver/amdxdna/aie2_pci.c (hunk 2 above)
Comment (mamin506) [reply src/driver/amdxdna/aie2_pci.c]: Done.

**Review by maxzhen — APPROVED** [review]

**Pull request merged**